MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs

```diff
@@ -6443,13 +6443,28 @@ private void strtol()
 
             var stringContainingLongs = Encoding.ASCII.GetString(Module.Memory.GetString(stringPointer, stripNull: true));
 
-            var longToParse = stringContainingLongs.Split(' ')[0];
-            var longToParseLength = longToParse.Length; //We do this as length might change with logic below
+            if (stringContainingLongs == "")
+            {
+                Registers.DX = 0;
+                Registers.AX = 0;
+
+                if (suffixPointer != IntPtr16.Empty)
+                    Module.Memory.SetPointer(suffixPointer, new IntPtr16(stringPointer.Segment, stringPointer.Offset));
+                
+                return;
+            }
+
+            var longToParse = stringContainingLongs.Split(' ', StringSplitOptions.RemoveEmptyEntries)[0];
+            var longToParseLength = longToParse.Length + (stringContainingLongs.Length - stringContainingLongs.TrimStart(' ').Length); //We do this as length might change with logic below and add back in leading spaces
 
-            if (longToParseLength == 0)
+            if (longToParseLength == 0 || !longToParse.Any(char.IsDigit))
```

Comment:
`char.IsDigit` likely isn't safe since hex values can have A-F for example

Reply:
I added a test for 0xFFFFF and also a max value test